### DISCUSSION TITLE
Crash while generating PDF file

### DISCUSSION THREAD

**Discussion by mgevher** [cefsharp/CefSharp#4353] [General]
- Version: 108.4.130
- Architecture: x64
- .Net: 4.7.2
- OS: Win10
- WPF
- Issue:

 We  want to export SVG image of height 70 inches and width 165 inches to a PDF file. Print settings for PrintToPdfAsync method are as follows:

![image](https://user-images.githubusercontent.com/44376702/209685820-fda24a12-f008-4dad-aead-08ea3a41e49e.png)

When version 108.4.130 is used, CefSharp.BrowserSubprocess memory size increases up to 12 Gb and  CefSharp.BrowserSubprocess crashes.
When previous version like 86.0.24 is used, CefSharp.BrowserSubprocess memory size increases up to 4 Gb and PDF is created successfully.

Callstack before the crash:

![MicrosoftTeams-image](https://user-images.githubusercontent.com/44376702/209686898-a92079fd-f084-4df0-8ee8-1a849bd53f73.png)

**Comment by amaitland** [comment]
Converted to [Discussion](https://github.com/cefsharp/CefSharp/discussions) as this is unlikely a bug that can be fixed in `CefSharp`. The actual code to generate a PDF is within `Chromium` itself.

> We want to export SVG image of height 70 inches and width 165 inches to a PDF file.

Have you tried generating a PDF with those dimensions in `Chrome`? Considering the standard size is 8.5 x 11 inches it's unlikely `Chrome` tests generating PDF that large.

If generating a `PDF` using `Chrome` works via the Print dialog then you'll need to find a way to use the [Page.PrintToPdf](https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-printToPDF) method provided by the devtools protocol.

You can potentially use https://github.com/hardkoded/puppeteer-sharp for this, you'd need to make sure the versions you were testing with match (which they won't by default).

You can also try version `107` of `CefSharp` as it uses a slightly different `Chromium` code path for generating a `PDF`. #4311 has link to the relevant `CEF` issue discussing the internal change.

> Print settings for PrintToPdfAsync method are as follows:

In future please provide actual formatted code rather than an image, I cannot copy and paste an image.

> Callstack before the crash:

Again, in future please provide formatted text instead of an image, cannot copy and paste from the image.

The call stack is deep within `Chromium` code, so likely that's where this will need to be fixed.

**Comment by amartens181** [comment]
I have yet to find the exact cause and report a bug, but Chrome has been running out of memory when attempting to generate PDFs with a few hundred pages with SVGs.
It may have to do with [LayoutNG printing](https://developer.chrome.com/blog/chrome-108-beta/#:~:text=This%20feature%20has%20been%20added%20due%20to%20Chrome%20108%20adding%20support%20for%20LayoutNG%20printing) or other recent rendering engine changes.
